pytorch_lightning/plugins/training_type/ipu.py

```diff
@@ -185,27 +183,21 @@ def _convert_to_poptorch_loader(
         dataloader = poptorch.DataLoader(**dl_kwargs, options=opts)
         return dataloader
 
-    @property
-    def accumulate_grad_batches(self) -> int:
-        return self._original_accumulate_grad_batches
-
     def _handle_gradient_accumulation_steps(self) -> None:
         """Override the trainer.accumulation_scheduler to act as ``accumulate_grad_batches=1`` if gradient
         accumulation has been set.
 
         ``optimizer_step`` will be called on every batch, and the IPU will handle grad accumulation internally.
         """
-        accumulate_grad_batches = self.lightning_module.trainer.accumulate_grad_batches
-        if not isinstance(accumulate_grad_batches, int):
+        accumulation_scheduler = self.lightning_module.trainer.accumulation_scheduler
+
+        if accumulation_scheduler.epochs != [0]:
             raise MisconfigurationException(
-                "IPUs currently only support `Trainer.accumulate_grad_batches` being an integer."
-                f" Received {accumulate_grad_batches}"
+                "IPUs currently does not support different `accumulate_grad_batches` at different epoch."
             )
-        # save the original value which will be used to update the global step progress
-        self._original_accumulate_grad_batches = accumulate_grad_batches
-        if accumulate_grad_batches > 1:
-            # TODO(@tchaton): Add support for accumulate_grad_batches being a dictionary
-            self.lightning_module.trainer.accumulation_scheduler = GradientAccumulationScheduler({0: 1})
+
+        # TODO(@tchaton): Add support for accumulate_grad_batches being a dictionary
+        accumulation_scheduler.scheduling.update({0: 1})
```

Comment:
I believe, dict can be supported but if we have multiple accumulation factor at different epoch.. then it's a problem.. so updated it accordingly along with the error message.